controllers/apps/transformer_rbac.go

```
@@ -58,7 +60,26 @@ func (c *RBACTransformer) Transform(ctx graph.TransformContext, dag *graph.DAG)
 		return err
 	}
 
-	for _, compSpec := range cluster.Spec.ComponentSpecs {
+	componentSpecs := make([]appsv1alpha1.ClusterComponentSpec, 0, 1)
+	compSpecMap := cluster.Spec.GetDefNameMappingComponents()
+	for _, compDef := range clusterDef.Spec.ComponentDefs {
+		comps := compSpecMap[compDef.Name]
+		if len(comps) == 0 {
+			// if componentSpecs is empty, it may be generated from the cluster template and cluster.
+			reqCtx := ictrlutil.RequestCtx{
+				Ctx: transCtx.Context,
+				Log: log.Log.WithName("rbac"),
+			}
+			synthesizedComponent, err := component.BuildComponent(reqCtx, nil, cluster, transCtx.ClusterTemplate, transCtx.ClusterDef, &compDef, nil)
+			if err != nil || synthesizedComponent == nil {
```

should not ignore the error